toqito/matrix_props/is_k_incoherent.py

```diff
@@ -0,0 +1,94 @@
+"""Checks if the matrix is $k$-incoherent."""
+
+from itertools import combinations
+
+import cvxpy as cp
+import numpy as np
+
+from toqito.matrices import comparison
+from toqito.matrix_props import is_positive_semidefinite, is_square
+
+
+def is_k_incoherent(mat: np.ndarray, k: int, tol: float = 1e-15) -> bool:
+    r"""Determine whether a quantum state is k-incoherent :cite:`Johnston_2022_Absolutely`.
+
```

Comment:
We should summarize what k incoherence is.

Reply:
Yes, definitely! 55a368f0d9211be1cca7bdc1e8df80986fa87ef7